modules/core/src/main/java/org/apache/ignite/cluster/ClusterState.java

```diff
@@ -29,4 +30,43 @@
 
     /** Cluster activated. Cache read operation allowed, Cache data change operation aren't allowed. */
     ACTIVE_READ_ONLY;
+
+    /** Enumerated values. */
+    private static final ClusterState[] VALS = values();
+
+    /**
+     * Efficiently gets enumerated value from its ordinal.
+     *
+     * @param ord Ordinal value.
+     * @return Enumerated value or {@code null} if ordinal out of range.
+     */
+    @Nullable public static ClusterState fromOrdinal(int ord) {
+        return ord >= 0 && ord < VALS.length ? VALS[ord] : null;
+    }
+
+    /**
+     * @param state Cluster state
+     * @return {@code True} if cluster in given cluster {@code state} is activated and {@code False} otherwise.
+     */
+    public static boolean active(ClusterState state) {
+        return state != INACTIVE;
+    }
+
+    /**
+     * @param state1 First given state.
+     * @param state2 Second given state.
+     * @return Lesser of given states. The order: {@link #ACTIVE} > {@link #ACTIVE_READ_ONLY} > {@link #INACTIVE}.
+     */
+    public static ClusterState lesserOf(ClusterState state1, ClusterState state2) {
+        if (state1 == state2)
+            return state1;
+
+        if (state1 == INACTIVE || state2 == INACTIVE)
+            return INACTIVE;
+
+        if (state1 == ACTIVE_READ_ONLY || state2 == ACTIVE_READ_ONLY)
+            return ACTIVE_READ_ONLY;
+
+        throw new IllegalArgumentException("Unknown cluster states. state1: " + state1 + ", state2: " + state2);
```

Comment:
how could we get this exception ? redundant code.

Reply:
Now neither. But in the future, we could have other modes.